packages/camera/example/android/app/src/main/java/io/flutter/plugins/cameraexample/MainActivity.java

```diff
@@ -1,13 +1,22 @@
 package io.flutter.plugins.cameraexample;
 
-import android.os.Bundle;
-import io.flutter.app.FlutterActivity;
-import io.flutter.plugins.GeneratedPluginRegistrant;
+import androidx.annotation.NonNull;
+import io.flutter.embedding.android.FlutterActivity;
+import io.flutter.embedding.engine.FlutterEngine;
+import io.flutter.embedding.engine.plugins.shim.ShimPluginRegistry;
+import io.flutter.plugins.camera.CameraPlugin;
+import io.flutter.plugins.pathprovider.PathProviderPlugin;
+import io.flutter.plugins.videoplayer.VideoPlayerPlugin;
 
 public class MainActivity extends FlutterActivity {
   @Override
-  protected void onCreate(Bundle savedInstanceState) {
-    super.onCreate(savedInstanceState);
-    GeneratedPluginRegistrant.registerWith(this);
+  public void configureFlutterEngine(@NonNull FlutterEngine flutterEngine) {
+    flutterEngine.getPlugins().add(new CameraPlugin());
+
+    ShimPluginRegistry shimPluginRegistry = new ShimPluginRegistry(flutterEngine);
```

Comment:
It might be a good idea to register these first. Given that they're required by CameraPlugin, it is conceivable that CameraPlugin could immediately attempt to utilize them, in which case this would fail. So it looks like an implementation detail that this happens to work. But up to you.

Reply:
That makes sense to me, but `GeneratedPluginRegistrant` registers them in this order. I thought it was better to match that.